File: Microsoft/solution2/solution2.py
def combinationSum3(self,k,n):
        result = []
        def backtracking(combination, n, start):
            if k == len(combination):
                if n == 0: result.append(combination.copy())
                return
            for i in range(start, 10):
                combination.append(i)
                backtracking(combination, n-i, i+1)
                combination.pop()
        backtracking([], n, 1)
        return result

File: Microsoft/solution2/solution2.cpp
#include<bits/stdc++.h>
using namespace std;

vector<vector<int>> result;
void backtracking(vector<int>& arr,int k,int n,int beg)
{
    if(k==arr.size())
    {
        if(n==0)
        {
            result.push_back(arr);
            return;
        }
    }
    for(int i=beg;i<10;i++)
    {
        arr.push_back(i);
        backtracking(arr,k,n-i,i+1);
        arr.pop_back();
    }
}

vector<vector<int>> combinationSum3(int k,int n)
{
    vector<int> series;
    backtracking(series,k,n,1);
    return result;
}